src/main/java/core/basesyntax/FigureSupplier.java

```diff
@@ -0,0 +1,68 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class FigureSupplier {
+    private static final double MAX_SIZE = 100;
+    private static final int FIGURES_TYPES_COUNT = FigureType.values().length;
+    private Random random = new Random();
+    private ColorSupplier colorSupplier = new ColorSupplier();
+    
+    public Figure getRandomFigure() {
+        int index = random.nextInt(FIGURES_TYPES_COUNT);
+        FigureType figureType = FigureType.values()[index];
+
+        return switch (figureType) {
+            case CIRCLE -> getRandomCircle();
+            case SQUARE -> getRandomSquare();
+            case RECTANGLE -> getRandomRectangle();
+            case RIGHT_TRIANGLE -> getRandomRightTriangle();
+            case ISOSCELES_TRAPEZOID -> getRandomIsoscelesTrapezoid();
+            default -> getDefaultFigure();
+        };
+    }
```

Comment:
Consider removing the 'default' case from the switch statement since you have covered all enum values. If new figures are added in the future, it would be better to handle them explicitly.